chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -125,38 +157,58 @@ def backward_gpu(self, inputs, grad_outputs):
+                    gx = (t == -1) ? 0 : (coeff[0] * (y - (c == t)));
+                ''',
+                'softmax_crossent_bwd')(
+                    y, cupy.expand_dims(t, 1), coeff, x.shape[1], n_unit)
+        else:
+            gx = cuda.elementwise(
+                'T y, raw T w, S t, raw T coeff, S n_channel, S n_unit',
+                'T gx',
+                '''
+                    const int c = (i / n_unit % n_channel);
+                    gx = t == -1 ? 0 : coeff[0] * (y - (c == t)) * w[t];
+                ''',
+                'softmax_crossent_bwd')(
+                    y, self.class_weight, cupy.expand_dims(t, 1), coeff,
+                    x.shape[1], n_unit)
         return gx, None
 
 
 def softmax_cross_entropy(
-        x, t, use_cudnn=True, normalize=True, cache_score=True):
+        x, t, use_cudnn=True, normalize=True, cache_score=True,
+        class_weight=None):
     """Computes cross entropy loss for pre-softmax activations.
 
     Args:
-        x (Variable): Variable holding a multidimensional array whose element
-            indicates unnormalized log probability: the first axis of the
-            variable represents the number of samples, and the second axis
+        x (~chainer.Variable): Variable holding a multidimensional array whose
+            element indicates unnormalized log probability: the first axis of
+            the variable represents the number of samples, and the second axis
             represents the number of classes. While this function computes
             a usual softmax cross entropy if the number of dimensions is equal
             to 2, it computes a cross entropy of the replicated softmax if the
             number of dimensions is greater than 2.
-        t (Variable): Variable holding an int32 vector of ground truth labels.
-            If ``t[i] == -1``, corresponding ``x[i]`` is ignored.
+        t (~chainer.Variable): Variable holding an int32 vector of ground truth
+            labels. If ``t[i] == -1``, corresponding ``x[i]`` is ignored.
         normalize (bool): If ``True``, this function normalizes the cross
             entropy loss across all instances. If ``False``, it only
             normalizes along a batch size.
         cache_score (bool): When it is ``True``, the function stores result
             of forward computation to use it on backward computation. It
             reduces computational cost though consumes more memory.
+        class_weight (~numpy.ndarray or ~chainer.cuda.cupy.ndarray): An array
+            that contains constant weights that will be multiplied with the
+            loss values along with the second dimension. The shape of this
+            array should be ``(x.shape[1],)``. If this is not ``None``, each
+            class weight ``class_weight[i]`` is actually multiplied to the
+            corresponding log-softmax output ``y[i]`` before calculating the
+            actual loss value.
```

Comment:
(I start another thread since the old one is collapsed into the outdated diff)
OK, I should've written `t[i]==j` instead of `y[i]==j`. I maybe now understand your point, so then you should clarify what `y` is and its shape, and write `y[:, i]` instead of `y[i]` (if y is log(softmax(x))).